CodeforAustralia/council-emissions-calculator-spike
# Add alt text to all images of the main flow

## Overview of the Pull Request:
Add Alt text to the images on the main survey pages.

Quite interesting research:

1. Started from adding alt text as normal: `alt="XXXXX`, it didn't work. 
2. Tried importing Image from next js:

```js
import Image from 'next/image';

// read in the docs that we have to asign width and height otherwise it might not render as image
<Image src={Q1Cloud} alt="Test Alt Text" width={100} height={100} />
```
it didn't work either , didn't render at all, forget about alt text

3. Tried importing `Img` or`Image` from Chakra and use like this:

```js
// tried with Img then Image separately
import {
  Img,
  Image
} from "@chakra-ui/react";

<Img as={Cloud} alt="Test" />
```
Img was rendered on the page but no alt text applied.

Realised the main issue is that the image is svg, so found another way to add alt-text to icons and use this instead: https://chakra-ui.com/docs/components/icon-button which helpts the screen readers to `give meaning to the button`. That's exactly what we need.

Tested with the scheen reader and it works:

![image](https://user-images.githubusercontent.com/88268603/224533819-17a7d1ab-9cce-4b58-8176-3bd4eb74d559.png)

## link to Trello card or Github issue
Trello: https://trello.com/c/rJeHHY4I
Git Issue: #218 

## How Has This Been Tested?
- Click on inspect and check if the source code has `aria-label` attribure for images;
- Use the screen reader above (fun experience) and click on the image, sometimes there is a lag when it hasn't finished reading previous text, so click a few times;

## Pull Request Checklist:

Make sure the following checkboxes`[ ]` are checked with `x` before sending your PR -- thank you!
